sympy/assumptions/handlers/order.py

```diff
@@ -66,32 +66,88 @@ def Add(expr, assumptions):
         if r is not True:
             return r
 
-        nonpos = 0
-        for arg in expr.args:
-            if ask(Q.negative(arg), assumptions) is not True:
-                if ask(Q.positive(arg), assumptions) is False:
-                    nonpos += 1
-                else:
-                    break
-        else:
-            if nonpos < len(expr.args):
-                return True
+
+        args = [arg for arg in expr.args if not ask(Q.zero(arg), assumptions)]
+        if not args:
+            return False
+
+        inf = neg_inf = pos_inf = neg = npos = nneg = False
+
+        for arg in args:
+            if pos_inf and neg_inf:
+                return None
+
+            _inf = ask(Q.infinite(arg), assumptions)
+            if _inf:
+                inf = True
```

Comment:
This is always false, because `Q.real(arg)` is checked to be true above, and the docs for `Q.real` say that the infinities are not included.

Note that if `inf` is always false, a lot of your code can be removed.

Reply:
This part of the code is done by Shikhar. @asmeurer @czgdp1807 Should I keep it or remove it?

Reply:
Noted. I am waiting for a mentor to confirm. @oscarbenjamin Shall I remove all of the code involving inf == True?

Reply:
These code were written before Q.real was changed to mean finite values only. Q.extended_real would behave similarly.

Reply:
This seems to back up oscar's claim that most of the stuff here should be translated to `extended_*`.